File: 프로그래머스/2/42747. H－Index/H－Index.py
# 내림차순으로 정렬하고, 인용수가 논문수랑 같아지는 지점 체크(논문수 1부터 증가, idx 이용)
def solution(citations):
  answer = 0
  citations.sort(reverse=True)
  for cnt, citation in enumerate(citations):
    if citation >= cnt+1:
      answer = cnt+1
  return answer

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
  var answer = 0;

  // 내림차순 정렬
  citations.sort((a, b) => b-a);

  [...citations].forEach((v, i) => {
    if(v >= i+1) answer = i+1
  });
  return answer;
}